src/lib/loadInstructions.js

```diff
@@ -10,47 +21,116 @@ function readYamlFile(p) {
   return yaml.load(txt);
 }
 
+/*
+Parse a match bit string (e.g., "0100000----------101-----0010011")
+into an array of single characters, each representing one bit.
+
+Returns null if the string isn’t 16 or 32 bits long.
+*/
 function parseMatchBits(matchStr) {
-  if (!matchStr || matchStr.length !== 32) return null;
+  if (!matchStr) return null;
+  if (matchStr.length !== 32 && matchStr.length !== 16) return null;
   return matchStr.split("");
 }
 
+/*
+Parse a bitfield location string into numeric segments.
+Example:
+  "31-25|11-7" → [ {from:31,to:25}, {from:11,to:7} ]
+
+Each segment defines one contiguous region of bits, and fields can have
+multiple disjoint segments (like immediates in S- or B-type instructions).
+*/
+function parseLocationSegments(loc) {
+  if (loc === undefined || loc === null) return [];
+  return String(loc)
+    .split("|")
+    .map(part => part.trim())
+    .filter(Boolean)
+    .map(part => {
+      const [hiStr, loStr] = part.split("-").map(n => n.trim());
+      const hi = parseInt(hiStr, 10);
+      const lo = loStr !== undefined && loStr !== "" ? parseInt(loStr, 10) : hi;
+      if (Number.isNaN(hi) || Number.isNaN(lo)) return null;
+      return {
+        from: Math.max(hi, lo),
+        to: Math.min(hi, lo)
+      };
+    })
+    .filter(Boolean)
+    .sort((a, b) => (b.from - a.from) || (b.to - a.to));
+}
+
+
+/*
+Extract all variable and constant fields from an instruction definition.
+
+This function handles both flat encodings and multi-variant encodings
+(e.g., RV32 / RV64). It returns an array of field descriptors like:
+
+  [
+    { label: "funct7=0100000", from: 31, to: 25, width: 7, kind: "const" },
+    { label: "rs2", from: 24, to: 20, width: 5, kind: "var" },
+    ...
+  ]
+
+These field objects describe what appears at each bit position.
+*/
 function computeFields(doc) {
   const fields = [];
   if (!doc.encoding) return fields;
-  const match = parseMatchBits(doc.encoding.match || "");
-  const vars = Array.isArray(doc.encoding.variables) ? doc.encoding.variables : [];
+
+  let enc = doc.encoding;
+  if (!enc.match && !enc.variables) {
+    // pick RV32 first, otherwise first available sub-encoding
+    const variants = Object.keys(enc); // e.g., ["RV32", "RV64"]
+    const chosenKey = variants.includes("RV32") ? "RV32" : variants[0];
+    enc = enc[chosenKey] || {};
+  }
```

Comment:
This could also use a brief comment explaining what `enc` is supposed to be. Does it only _sometimes_ include `RV32` and `RV64` keys, for example? That would be worth noting.